### PR TITLE
Add Circom language support

### PR DESCRIPTION
**Pull request by cryptopapi997**
Added the Circom language

## Checklist:
- [x] **I am adding a new language.**
  - [x] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension: [>17k uses](https://github.com/search?utf8=%E2%9C%93&type=Code&ref=searchresults&q=extension%3Acircom)
  - [x] I have included a real-world usage sample for all extensions added in this PR:
      - [xor3, gpl3](https://github.com/iden3/circomlib/blob/master/circuits/sha256/xor3.circom)
      - [binsum, gpl3](https://github.com/iden3/circomlib/blob/master/circuits/binsum.circom)
      - [switcher, gpl3](https://github.com/iden3/circomlib/blob/master/circuits/switcher.circom)
    
  - [x] I have included a syntax highlighting grammar: Only available grammar I could find was [this one by the developers of circom](https://github.com/iden3/circom-highlighting-vscode), but it is currently unlicensed.
  - [x] I have updated the heuristics to distinguish my language from others using the same extension.

### CODE REVIEW

**Comment by Alhadis** [comment]
> It is GPL3, however

That's a deal-breaker. We only accept grammars released under [permissive licenses](https://github.com/github/linguist/blob/c1c34e5260797b4d598f5ec76f19723bfc5a1894/vendor/licenses/config.yml), which rules out copyleft licenses like GPL completely.

**Comment by Alhadis** [comment]
> Can we still merge this PR without a grammar for now nonetheless?

Yes. You can even reappropriate another language's grammar if it happens to provide decent-enough highlighting.

**Comment by krlosMata** [comment]
I will check other licenses that fits the requirements

**Comment by krlosMata** [comment]
> > It is GPL3, however
> 
> That's a deal-breaker. We only accept grammars released under [permissive licenses](https://github.com/github/linguist/blob/c1c34e5260797b4d598f5ec76f19723bfc5a1894/vendor/licenses/config.yml), which rules out copyleft licenses like GPL completely.

I have change the license to MIT in https://github.com/iden3/circom-highlighting-vscode, so it fulfills the specification

**Comment by cryptopapi997** [comment]
Awesome! Currently out, but will add the grammar when I'm back :+1:

**Comment by cryptopapi997** [comment]
Added the circom grammar, feel free to have a look again @Alhadis

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -1016,6 +1016,16 @@ ChucK:
   codemirror_mode: clike
   codemirror_mime_type: text/x-java
   language_id: 57
+Circom:
+  type: programming
+  ace_mode: java
+  codemirror_mode: clike
+  codemirror_mime_type: text/x-java
```
Comment (lildude) [inline lib/linguist/languages.yml:1023]: These don't seem right to me. The `codemirror` options are optional so if in doubt, remove and leave `ace_mode` as `text`:

```suggestion
  ace_mode: text
```

File: lib/linguist/languages.yml [hunk 2]
```diff
@@ -1016,6 +1016,16 @@ ChucK:
...
+  extensions:
+  - ".circom"
+  color: "#707575"
+  tm_scope: none
```
Comment (lildude) [inline lib/linguist/languages.yml:1027]: This needs to be set to the scope in the grammar:

```suggestion
  tm_scope: source.circom
```

**Review by lildude — CHANGES_REQUESTED** [review]

**Comment by cryptopapi997** [comment]
Thanks for the review, added your changes

**Review by lildude — APPROVED** [review]
LGTM. Thanks.

Note: this PR will not be merged until close to when the next release is made. See [here](https://github.com/github/linguist/blob/master/docs/troubleshooting.md#my-linguist-pr-has-been-merged-but-github-doesnt-reflect-my-changes) for more details.

**Pull request merged**